src/config-file-schema.ts

```diff
@@ -30,6 +30,7 @@ function isErrnoException(error: any): error is NodeJS.ErrnoException {
 export function ConfigFileSchema<T>(opts: {
   path: string;
   schema: JSONSchemaType<T>;
+  validate?: ValidateFunction;
```

Comment:
Since the schema is only used to generate the validate function, might it be cleaner to just require the validate function as input?

Reply:
The reasoning behind optional field was that we wanted to make sure we don't impact the existing implementations and uses of configFileSchema, especially since any schema that doesn't have references will work perfectly fine without provided validation function; additionally, it provides flexibility - if you don't have an exported validation function, you can just provide schema and it'll work.

With that being said, I think it's a bit subjective, and I agree that it might be cleaner to just require the validate function, should we then just drop schema completely?

Reply:
I added that functionality specifically for you to use for the model configurations, so if it doesn't meet the needs we should just make the change. I think there's only one other usage of it in the Device Agent which will be an easy update.